### PR TITLE
Fix setting NULL in profile definition

### PR DESCRIPTION
**Pull request by vitlibar**
### Changelog category:
- Bug Fix

### Changelog entry:
`CREATE USER u1 SETTINGS a=NULL` now works correctly.

### CODE REVIEW

File: tests/queries/0_stateless/01418_custom_settings.sql [hunk 1]
```diff
@@ -37,11 +41,23 @@ SELECT name, value FROM system.settings WHERE name = 'custom_f' SETTINGS custom_
...
+
+SELECT '--- null type ---';
+SELECT getSetting('custom_null') as v, toTypeName(v) SETTINGS custom_null = NULL;
+SELECT name, value FROM system.settings WHERE name = 'custom_null' SETTINGS custom_null = NULL;
+
+SET custom_null = NULL;
+SELECT getSetting('custom_null') as v, toTypeName(v);
+SELECT name, value FROM system.settings WHERE name = 'custom_null';
+
+CREATE SETTINGS PROFILE s2_01418 SETTINGS custom_null = NULL;
```
Comment (vitlibar) [inline tests/queries/0_stateless/01418_custom_settings.sql:61]: This line earlier stored a settings profile which couldn't be parsed:
```
CREATE SETTINGS PROFILE s2_01418 SETTINGS custom_null
```

**Comment by robot-ch-test-poll4** [comment]
This is an automated comment for commit a4694ac1858890bcd3d35547cf5c4417252933c9 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49831/a4694ac1858890bcd3d35547cf5c4417252933c9/ci_running.html)
The overall status of the commit is 🟢 success

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>Style Check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td>🟢 success</td></tr>
</table>

**Review by yakov-olkhovskiy — APPROVED** [review]
I'm not sure how to interpret Flaky tests showing failure for this test...

**Comment by vitlibar** [comment]
> I'm not sure how to interpret Flaky tests showing failure for this test...

It says "settings profile `s1_01418`: cannot insert because settings profile `s1_01418` already exists".
It happened because that flaky check ran this test in parallel with itself.
That shouldn't happen in master, so we can just ignore it here.

**Pull request merged**